llvm/docs/CodeReview.rst

```diff
@@ -150,14 +150,21 @@ almost always associated with a message containing the text "LGTM" (which
 stands for Looks Good To Me). Only approval from a single reviewer is required.
 
 When providing an unqualified LGTM (approval to commit), it is the
-responsibility of the reviewer to have reviewed all of the discussion and
+responsibility of the reviewer to have reviewed all of the prior discussion and
 feedback from all reviewers ensuring that all feedback has been addressed and
 that all other reviewers will almost surely be satisfied with the patch being
 approved. If unsure, the reviewer should provide a qualified approval, (e.g.,
 "LGTM, but please wait for @someone, @someone_else"). You may also do this if
 you are fairly certain that a particular community member will wish to review,
 even if that person hasn't done so yet.
 
+If new comments are posted after the patch has been approved (but not yet
```

Comment:
This advise talks about action the reviewer should take, that seems backwards here.  Unless I'm misreading, the problematic situation here is a prior LGTM followed by changes by a *different* reviewer and the author landing the patch without addressing those changes.

I'd suggest something along the lines of the following:
Only approval from a single reviewer is required; however the patch author is always responsible for making sure that all feedback on a review has been properly addressed or responded to.  If additional feedback is provided after acceptance (by the same reviewer or another), the author should use their best judgement in deciding whether that feedback can be incorporated into the change without comment (say a typo) or requires further review discussion.  Similarly, the patch author is responsible for incorporating post commit feedback on the change.  

I'd put this paragraph just above the one talking about unqualified LGTM

Reply:
>  Unless I'm misreading, the problematic situation here is a prior LGTM followed by comments by a different reviewer and the author landing the patch without addressing those comments.

Yes

>  (edit: for typo fix)

No, for more substantial changes. For "typos", I added an exception.

>  If additional feedback is provided after acceptance (by the same reviewer or another), the author should use their best judgement in deciding whether that feedback can be incorporated into the change without comment (say a typo) or requires further review discussion.

To me this sounds that any comments "post LGTM" are "second category" (i.e. "for the author to decide whether to address them or not" vs "need to be addressed"). I'm suggesting that we should treat such comments as any other comments prior to LGTM (i.e., "need to be addressed"). I feel that that otherwise we are discouraging people from reading/commenting on patches post LGTM.

Reply:
> > Unless I'm misreading, the problematic situation here is a prior LGTM followed by comments by a different reviewer and the author landing the patch without addressing those comments.
> 
> Yes
> 
> > (edit: for typo fix)
> 
> No, for more substantial changes. For "typos", I added an exception.

(You're responding to a note I made about editting my text to fix a typo...)
 
> > If additional feedback is provided after acceptance (by the same reviewer or another), the author should use their best judgement in deciding whether that feedback can be incorporated into the change without comment (say a typo) or requires further review discussion.
> 
> To me this sounds that any comments "post LGTM" are "second category" (i.e. "for the author to decide whether to address them or not" vs "need to be addressed"). I'm suggesting that we should treat such comments as any other comments prior to LGTM (i.e., "need to be addressed"). I feel that that otherwise we are discouraging people from reading/commenting on patches post LGTM.

If you have suggested wording, feel free to suggest.  I don't read this the way you do, particular following the prior sentence that you trimmed.

Reply:
Hi @preames , thanks for that suggestion and sorry for the delay - I had to step away and re-think this a bit. I do want to incorporate your feedback, but also want to make sure we avoid duplication and keep this document succinct. So I trimmed your suggestion a bit as per the comments below (hope this won't come across as nit-picking).

**Point 1:**
> [From Phillip's suggestion]
> Only approval from a single reviewer is required; 

This point is already mentioned earlier in the document: https://github.com/llvm/llvm-project/blob/c2063de1593610eda0f4de33c3b89324642ed54c/llvm/docs/CodeReview.rst?plain=1#L148-L150
 
**Point 2:**
> [From Phillip's suggestion]
> however the patch author is always responsible for making sure that all feedback on a review has been properly addressed or responded to. 

Isn't this a contradicting one existing policy?
https://github.com/llvm/llvm-project/blob/c2063de1593610eda0f4de33c3b89324642ed54c/llvm/docs/CodeReview.rst?plain=1#L152-L159

Specifically:

> it is the responsibility of the reviewer to have reviewed all of the discussion and feedback from all reviewers ensuring that all feedback has been addressed and that all other reviewers will almost surely be satisfied with the patch being approved.

I'd just leave this out.

**Point 3:**

> [From Phillip's suggestion]
> Similarly, the patch author is responsible for incorporating post commit feedback on the change.

This is already covered here: https://github.com/llvm/llvm-project/blob/c2063de1593610eda0f4de33c3b89324642ed54c/llvm/docs/CodeReview.rst?plain=1#L46-L49

**Point 4:**

Which leaves this bit from Phillip's original suggestion (emphasis mine):
> [From Phillip's suggestion]
If additional feedback is provided after acceptance (by the same reviewer or another), **the author should use their best judgement** in deciding whether that feedback can be incorporated into the change without comment (say a typo) or requires further review discussion. 

This a bit softer requirement compared to what I added "these need to be addressed" and sounds more along the lines of what @kuhar would prefer? Let me use that, but let me add small clarification at the end. Let me know what you think.

Reply:
You have seemingly managed to complete miss the spirit of my original suggestion, and are misreading the current policy in my view.  If the document implies that the author is not responsible for incorporating feedback (because that's only the reviewers responsibility), the document is simply wrong.  I don't read it that way, but if you do, stop and fix that wording before working on this smaller item.

Reply:
> You have seemingly managed to complete miss the spirit of my original suggestion

That's not intentional. Would you be able to share a diff so that I better understand what you had in mind? 

> If the document implies that the author is not responsible for incorporating feedback

I don’t believe the document implies that. I think it’s fairly clear that:

1. The author is responsible for addressing/incorporating the feedback, which is already covered under "Acknowledge All Reviewer Feedback" (so perhaps we can leave that as is).
2. The reviewer, when posting an unqualified LGTM, is responsible for ensuring that all feedback has been addressed (i.e., “I've reviewed all comments and confirm they’ve been addressed").

The existing wording covers those points well. The part that is unclear IMO is what happens **after LGTM** and **before commit** when new comments are posted. Hopefully, the additional paragraph I proposed clarifies that.